pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -433,14 +433,19 @@ func (c *InstallerController) manageInstallationPods(ctx context.Context, operat
 
 		klog.Infof("%s and needs new revision %d", nodeChoiceReason, revisionToStart)
 
-		newCurrNodeState := currNodeState.DeepCopy()
-		newCurrNodeState.TargetRevision = revisionToStart
-		if newCurrNodeState.LastFailedRevision != revisionToStart {
-			newCurrNodeState.LastFailedRevisionErrors = nil
-			newCurrNodeState.LastFailedCount = 0
-			newCurrNodeState.LastFailedTime = nil
-		} else if newCurrNodeState.LastFailedCount == 0 {
-			newCurrNodeState.LastFailedCount = 1
+		var newCurrNodeState *operatorv1.NodeStatus
+		if currNodeState.LastFailedRevision != revisionToStart {
+			newCurrNodeState = &operatorv1.NodeStatus{
+				NodeName:        currNodeState.NodeName,
+				CurrentRevision: currNodeState.CurrentRevision,
+				TargetRevision:  revisionToStart,
+			}
+		} else {
+			newCurrNodeState = currNodeState.DeepCopy()
+			newCurrNodeState.TargetRevision = revisionToStart
+			if newCurrNodeState.LastFailedCount == 0 {
+				newCurrNodeState.LastFailedCount = 1
+			}
```

Comment:
this logic is questionable, probably wrong. The following commit get rid of it. Here, I only rewrite.